lib/chip/service.rb

```diff
@@ -44,6 +55,26 @@ def token
       end
     end
 
+    def with_monitoring_and_error_handling(&)
+      with_monitoring(2, &)
+    rescue => e
+      handle_error(error: e)
+    end
+
+    def handle_error(error:, message: nil)
+      raise case error
+            when Common::Client::Errors::ClientError
+              Raven.extra_context body: error.body, status: error.status
```

Comment:
Instead of using Raven.extra_context, please `include SentryLogging` to `Service` class & call `log_exception_to_sentry`